automatedtests/src/androidTest/java/com/mux/stats/sdk/muxstats/automatedtests/BandwidthMetricTests.java

```diff
@@ -0,0 +1,399 @@
+        requestCompletedEventIndex, requestJson, segmentUrl);
+    checkStringValueForEvent("REQUEST_HOSTNAME", BandwidthMetricData.REQUEST_HOSTNAME,
+        requestCompletedEventIndex, requestJson, "localhost");
+    if (!requestJson.has(BandwidthMetricData.REQUEST_ERROR)) {
+      fail("Request failed missing REQUEST_ERROR field for event index: "
+          + requestCompletedEventIndex);
+    }
+    if (!requestJson.has(BandwidthMetricData.REQUEST_ERROR_CODE)) {
+      fail("Request failed missing REQUEST_ERROR_CODE field for event index: "
+          + requestCompletedEventIndex);
+    }
+    if (!requestJson.has(BandwidthMetricData.REQUEST_ERROR_TEXT)) {
+      fail("Request failed missing REQUEST_ERROR_TEXT field for event index: "
+          + requestCompletedEventIndex);
+    }
+  }
+
+  private void checkRequestCanceledEvent(JSONObject requestJson,
+      boolean expectingManifest,
+      SegmentStatistics segmentStat,
+      int requestCompletedEventIndex,
+      String segmentUrl)
+      throws Exception {
+    checkManifestValue(requestJson, expectingManifest);
+    checkLongValueForEvent(
+        "REQUEST_RESPONSE_START", BandwidthMetricData.REQUEST_RESPONSE_START,
+        requestCompletedEventIndex, requestJson, segmentStat.getSegmentRequestedAt(),
+        ERROR_MARGIN_FOR_REQUEST_LATENCY);
+    checkLongValueForEvent(
+        "REQUEST_RESPONSE_END", BandwidthMetricData.REQUEST_RESPONSE_END,
+        requestCompletedEventIndex, requestJson, segmentStat.getSegmentRespondedAt(),
+        ERROR_MARGIN_FOR_REQUEST_LATENCY);
+    if (!requestJson.has(BandwidthMetricData.REQUEST_CANCEL)) {
+      fail("Request canceled missing REQUEST_CANCEL field for event index: " + requestCompletedEventIndex);
+    }
+    checkStringValueForEvent("REQUEST_URL", BandwidthMetricData.REQUEST_URL,
+        requestCompletedEventIndex, requestJson, segmentUrl);
+    checkStringValueForEvent("REQUEST_HOSTNAME", BandwidthMetricData.REQUEST_HOSTNAME,
+        requestCompletedEventIndex, requestJson, "localhost");
+  }
+
+  private void checkRequestCompletedEvent(JSONObject requestJson,
+      boolean expectingManifest,
+      SegmentStatistics segmentStat,
+      int mediaSegmentIndex,
+      int qualityLevel,
+      int requestCompletedEventIndex,
+      String fileNameHeaderValue,
+      long requestNetworkDelay
+  ) throws Exception {
+    if(checkManifestValue(requestJson, expectingManifest)) {
+      checkManifestSegment(requestCompletedEventIndex, requestJson, segmentStat);
+    } else {
+      checkNetworkDelay(requestJson, requestNetworkDelay, requestCompletedEventIndex);
+      checkRequestCompletedMediaSegment(requestCompletedEventIndex, requestJson,
+          mediaSegmentIndex, qualityLevel, fileNameHeaderValue, segmentStat);
+    }
+  }
+
+  private void checkNetworkDelay(JSONObject requestJson, long expectedNetworkDelay, int eventIndex) {
+    // I have no way of getting REQUEST_START, so I need to comment this
```

Comment:
Remove this unused commented-out code, but preserve this note about being unable to get `REQUEST_START`

Reply:
removed